src/dataconnect/schemaMigration.ts

```diff
@@ -62,6 +63,23 @@ export async function migrateSchema(args: {
     databaseId,
     /* linkIfNotConnected=*/ true,
   );
+
+  if (experiments.isEnabled("compatiblemode")) {
+    if (schema.primaryDatasource.postgresql?.schemaValidation) {
+      schema.primaryDatasource.postgresql.schemaValidation = "COMPATIBLE";
+    } else {
+      schema.primaryDatasource = {
+        postgresql: {
+          database: databaseId,
+          cloudSql: {
+            instance: instanceName,
+          },
+          schemaValidation: "COMPATIBLE",
+        },
+      };
+    }
+  }
+
```

Comment:
Looks great! Can you also add this into the `diffSchema` method as well? It might be worth extracting this into a helper method, to reduce duplication as well.